server/index.js

```diff
@@ -14,141 +15,82 @@ const loggingMiddleware = require('./logging');
   );
 
-  if (!dev) {
-    createServer({ port: 9153 }).then(() => console.log(`@promster/server started on port 9153.`));
-    server.use(loggingMiddleware());
-  }
+  server.use(
+    (() => {
+      const router = express.Router();
+      router.use('/txid', next());
+      router.use('/microblock', next());
+      router.use('/block', next());
+      router.use('/address', next());
+      return router;
+    })()
+  );
 
-  server.get('/', (req, res) => {
-    // since we don't use next's requestHandler, we lose compression, so we manually add it
-    server.use(compression());
-    renderAndCache(app)(req, res, '/');
-  });
+  let routes = [];
 
-  server.get('/*.js', (req, res, next) => {
-    req.url = req.url + '.br';
-    res.set('Content-Encoding', 'br');
-    res.set('Content-Type', 'application/javascript; charset=UTF-8');
-    next();
-  });
+  // Store all the registered express routes for usage with metrics reporting
+  routes = expressListEndpoints(server).map(endpoint => ({
+    path: endpoint.path,
+    regexp: pathToRegexp(endpoint.path),
+  }));
 
-  server.get('/txid/:txid', (req, res) => {
-    server.use(compression());
-    const queryParams = { txid: req.params.txid };
-    const pagePath = '/txid/[txid]';
-    renderAndCache(app)(req, res, pagePath, queryParams);
+  // Manual route definitions for the proxied endpoints
+  routes.push({
+    path: '/txid/*',
+    regexp: /^\/txid(.*)/,
   });
-  server.get('/tx/:txid', (req, res) => {
-    server.use(compression());
-    const queryParams = { txid: req.params.txid };
-    const pagePath = '/txid/[txid]';
-    renderAndCache(app)(req, res, pagePath, queryParams);
+  routes.push({
+    path: '/microblock/*',
+    regexp: /^\/microblock(.*)/,
   });
-
-  server.get('/block/:hash', (req, res) => {
-    server.use(compression());
-    const queryParams = { hash: req.params.hash };
-    const pagePath = '/block/[hash]';
-    renderAndCache(app)(req, res, pagePath, queryParams);
+  routes.push({
+    path: '/block/*',
+    regexp: /^\/block(.*)/,
   });
-
-  server.get('/address/:principal', (req, res) => {
-    server.use(compression());
-    const queryParams = { principal: req.params.principal };
-    const pagePath = '/address/[principal]';
-    renderAndCache(app)(req, res, pagePath, queryParams);
+  routes.push({
+    path: '/address/*',
+    regexp: /^\/address(.*)/,
   });
 
-  server.get('*', (req, res, next) => {
+  if (!dev) {
+    const promMiddleware = createPrometheusMiddleware({
+      options: {
+        normalizePath: path => {
+          // Get the url pathname without a query string or fragment
+          // (note base url doesn't matter, but required by URL constructor)
+          try {
+            let pathTemplate = new URL(path, 'http://x').pathname;
+            // Match request url to the Express route
+            for (const pathRegex of routes) {
+              if (pathRegex.regexp.test(pathTemplate)) {
+                pathTemplate = pathRegex.path;
```

Comment:
Seems like it would be useful to console.log to see if this code path is actually triggered when hitting on of the explorer pages. I'm not sure if the `expressListEndpoints` function is actually helping much here given the defined express routes above (i.e. not sure if `router.use('/txid', next())` generates regex that will get matched correctly).

Although, even if the auto-magic express route detection matching doesn't work, it does look like the manually defined paths and regexes in the `routes` array would work.